catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/workspace/transformer/WorkspaceTransformation.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.metacard.workspace.transformer;
+
+import javax.annotation.Nullable;
+
+/**
+ * A representation of one key-value transformation for the {@link WorkspaceTransformer}.
+ * Implementations of this interface can be used to transform JSON-style key-value pairs into
+ * metacard attributes and vice versa, especially from and into workspace metacards.
+ *
+ * @param <M> the expected type of metacard values handled by this {@link WorkspaceTransformation}
+ * @param <J> the expected type of JSON-style key-value pair values handled by this {@link
+ *     WorkspaceTransformation}
+ */
```

Comment:
Should probably drop the experimental blurb in here. Is this an internal interface? If so we should also put it into a `.internal` package

Reply:
You're definitely right that I should add the "experimental" comment. I will add that.

I am not quite sure of the definition of "internal" interface here, but this interface is intended to be used in downstream projects in order to extend the functionality of the `WorkspaceTransformer`, so I doubt that it would be considered internal. Let me know if I'm wrong.